tests/integration/setup_environment.py

```diff
@@ -0,0 +1,99 @@
+from typing import Self
+import socket
+
+from testcontainers.core.container import DockerContainer
+from testcontainers.core.network import Network
+from testcontainers.core.waiting_utils import wait_container_is_ready, wait_for_logs
+from testcontainers.postgres import PostgresContainer
+
+from adcm_aio_client.core.errors import DockerContainerError
+
+postgres_image_name = "postgres:latest"
+adcm_image_name = "hub.adsw.io/adcm/adcm:develop"
+adcm_port_range = (8000, 8010)
+postgres_port_range = (5432, 5442)
+adcm_container_name = "test_adcm"
+postgres_name = "test_pg_db"
+db_user = "adcm"
+db_name = "adcm"
+db_password = "password"  # noqa: S105
+
+
+def find_free_port(start: int, end: int) -> int:
+    """Try to find a free port in the given range."""
+    for port in range(start, end):
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            if s.connect_ex(("127.0.0.1", port)) != 0:  # Port is free
+                return port
+    raise DockerContainerError(f"No free ports found in the range {start} to {end}")
+
+
+class ADCMPostgresContainer(PostgresContainer):
+    url: str
+
+    def __init__(self: Self, image: str) -> None:
+        super().__init__(image)
+
+    def setup_postgres(self: Self, username: str, password: str, adcm_db_name: str, container_network: Network) -> None:
+        postgres_port = find_free_port(postgres_port_range[0], postgres_port_range[1])
+        self.with_name(f"{postgres_name}_{postgres_port}")
+        self.with_bind_ports(postgres_port, postgres_port)
+        self.password = password
+        self.with_network(container_network)
+
+        self.start()
+        wait_container_is_ready(self)
+
+        init_db_cmds = (
+            f"CREATE USER {username} WITH ENCRYPTED PASSWORD '{password}'; "
+            f"CREATE DATABASE {adcm_db_name} OWNER {username};"
+        )
+        self.with_command(
+            f"PGPASSWORD={self.password} psql --username {self.username} --host 127.0.0.1 -c '{init_db_cmds}'"
+        )
+
+        wait_for_logs(self, "database system is ready to accept connections")
+
+        self.url = f"http://{self.get_container_host_ip()}:{self.get_exposed_port(postgres_port)}"
```

Comment:
I don't think we need to connect to http for postgres

Reply:
deleted